File: frontend/src/types/domain/table-monitor.ts
export type MonitoredDatabaseRole = 'business' | 'dataset' | 'usage-catalog' | 'stats' | 'codex-context-state'

export interface DatabaseStorageSnapshotSummary {
  databaseRole: MonitoredDatabaseRole
  databasePath: string
  sampledAt: string
  fileBytes?: number
  walBytes?: number
  shmBytes?: number
  freeBytes?: number
  tableCount?: number
}

export interface DatabaseStorageHistoryPoint {
  databaseRole: MonitoredDatabaseRole
  sampledAt: string
  fileBytes?: number
  walBytes?: number
  freeBytes?: number
  tableCount?: number
}

export interface TableStorageHistoryPoint {
  sampledAt: string
  rowCount?: number
  totalBytes?: number
}

export interface TableStorageOverviewSummary {
  databaseRole: MonitoredDatabaseRole
  tableName: string
  sampledAt: string
  tableKind?: string
  parentTableName?: string
  isPartition?: boolean
  isArchive?: boolean
  rowCount?: number
  tableBytes?: number
  indexBytes?: number
  indexToTableRatio?: number
  totalBytes?: number
  growthBytes1h?: number
  growthRows1h?: number
  growthBytes24h?: number
  growthRows24h?: number
}

export interface TableStorageOverview {
  sampledAt?: string
  databases: DatabaseStorageSnapshotSummary[]
  tables: TableStorageOverviewSummary[]
  page: number
  pageSize: number
  total: number
  hasMore: boolean
}

export interface NonBusinessDataCleanupResult {
  cutoffAt: string
  queued: boolean
  jobId?: string
  submittedAt?: string
  blockedReason?: string
}

File: backend-go/internal/store/port/tablemonitor.go
package port

import "context"

type MonitoredDatabaseRole string

const (
	MonitoredDatabaseRoleBusiness          MonitoredDatabaseRole = "business"
	MonitoredDatabaseRoleDataset           MonitoredDatabaseRole = "dataset"
	MonitoredDatabaseRoleUsageCatalog      MonitoredDatabaseRole = "usage-catalog"
	MonitoredDatabaseRoleStats             MonitoredDatabaseRole = "stats"
	MonitoredDatabaseRoleCodexContextState MonitoredDatabaseRole = "codex-context-state"
)

type ManagementDatabaseStorageSnapshot struct {
	DatabaseRole  MonitoredDatabaseRole
	DatabasePath  string
	SampledAt     string
	FileBytes     *int64
	WALBytes      *int64
	SHMBytes      *int64
	PageSize      *int64
	PageCount     *int64
	FreelistCount *int64
	UsedBytes     *int64
	FreeBytes     *int64
	TableCount    *int64
	IndexCount    *int64
}

type ManagementTableStorageSnapshot struct {
	DatabaseRole    MonitoredDatabaseRole
	TableName       string
	SampledAt       string
	TableKind       *string
	ParentTableName *string
	IsPartition     bool
	IsArchive       bool
	RowCount        *int64
	TableBytes      *int64
	IndexBytes      *int64
	TotalBytes      *int64
	PageCount       *int64
	IndexCount      int64
	GrowthBytes1H   *int64
	GrowthRows1H    *int64
	GrowthBytes24H  *int64
	GrowthRows24H   *int64
}

type ManagementTableStorageOverview struct {
	SampledAt string
	Databases []ManagementDatabaseStorageSnapshot
	Tables    []ManagementTableStorageSnapshot
}

type ManagementTableStorageHistoryInput struct {
	DatabaseRole MonitoredDatabaseRole
	TableName    string
	StartAt      string
	EndAt        string
	Limit        int
}

type ManagementDatabaseStorageHistoryInput struct {
	StartAt string
	EndAt   string
	Limit   int
}

type ManagementTableMonitorReader interface {
	GetManagementTableStorageOverview(ctx context.Context, limit int) (ManagementTableStorageOverview, error)
	ListManagementTableStorageHistory(ctx context.Context, input ManagementTableStorageHistoryInput) ([]ManagementTableStorageSnapshot, error)
	ListManagementDatabaseStorageHistory(ctx context.Context, input ManagementDatabaseStorageHistoryInput) ([]ManagementDatabaseStorageSnapshot, error)
}
